tools/pipelines/templates/include-publish-npm-package-steps.yml

```diff
@@ -82,32 +82,36 @@ steps:
       echo Tag: $tag
       cp .npmrc ~/.npmrc
       t1=0
-      for f in *.tgz
+      sortedPackages=$(cat $(Pipeline.Workspace)/pack/packagePublishOrder.txt)
+      for packageName in $sortedPackages
       do
-          npm publish $f $tag ${{ parameters.publishFlags }} 2>&1 | tee publish_log
-          if grep -q "npm ERR!" "publish_log"
-          then
-              if ! grep -q "You cannot publish over the previously published versions" "publish_log"
-              then
-                let t1+=1
-              else
-                echo Package has already been published.
-                local_f="${f}_local"
-                mv "$f" "$local_f"
-                package_name=$(grep -oP 'npm notice package: \K.*' "publish_log")
-                npm pack $package_name
-                if cmp -s "$f" "$local_f"
+          f=$(find -name "${packageName}-[0-9]*.tgz")
```

Comment:
Does that work with our internal version schema `2.0.0-internal.x.y.z`?

Reply:
Yes, since this is a glob pattern and not regex. This pattern expects a single number after the package name to get around some packages starting with the same names e.g. `a-b-2.0.0.tgz` and `a-b-c-2.0.0.tgz`. Then as long as the name ends in `.tgz`, the file will match.